src/phpDocumentor/Descriptor/Builder/Matcher.php

```diff
@@ -15,15 +15,15 @@ final class Matcher
     private $type;
 
     /**
-     * @param class-string<StaticT> $type
+     * @param class-string<SelfT> $type
      *
-     * @return static<StaticT>
+     * @return self<SelfT>
      *
-     * @template StaticT
+     * @template SelfT
      */
     public static function forType(string $type) : self
     {
-        return new static($type);
+        return new Matcher($type);
```

Comment:
please use `self` instead; makes for less changes when refactoring and is more expressive than re-using the classname.

Reply:
done :)